public-api/src/main/java/org/pmiops/workbench/google/GoogleAnalyticsServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.pmiops.workbench.google;
+
+import com.google.appengine.api.urlfetch.HTTPHeader;
+import com.google.appengine.api.urlfetch.HTTPMethod;
+import com.google.appengine.api.urlfetch.HTTPRequest;
+import com.google.appengine.api.urlfetch.HTTPResponse;
+import com.google.appengine.api.urlfetch.URLFetchService;
+import com.google.appengine.api.urlfetch.URLFetchServiceFactory;
+import com.google.common.annotations.VisibleForTesting;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@Service
+public class GoogleAnalyticsServiceImpl implements GoogleAnalyticsService {
+
+    private final HTTPHeader header =
+            new HTTPHeader("Content-Type", "application/x-www-form-urlencoded");
+    private URLFetchService urlFetchService;
+    private final Provider<WorkbenchConfig> configProvider;
+
+    @Autowired
+    public GoogleAnalyticsServiceImpl(Provider<WorkbenchConfig> configProvider) {
+        this.configProvider = configProvider;
+        setUrlFetchService(getUrlFetchService());
```

Comment:
This is a bit odd - can you not just inject it into the constructor like a normal dependency? Seems like your test would also work fine with this.

Note: outside of tests would also need to provide this somewhere - `AppEngineConfig` would probably make sense:

```
@Bean
URLFetchService urlFetchService() {
  return URLFetchServiceFactory.getURLFetchService();
}
```

Reply:
Thank you - I will look at that.

Reply:
Done